Rebus/Encryption/EncryptMessagesOutgoingStep.cs

```diff
@@ -37,10 +38,13 @@ public async Task Process(OutgoingStepContext context, Func<Task> next)
 
             var headers = transportMessage.Headers.Clone();
             var bodyBytes = transportMessage.Body;
-            var encryptedData = _encryptor.Encrypt(bodyBytes);
+            var encryptedData = await _encryptor.Encrypt(bodyBytes);
 
             headers[EncryptionHeaders.ContentEncryption] = _encryptor.ContentEncryptionValue;
             headers[EncryptionHeaders.ContentInitializationVector] = Convert.ToBase64String(encryptedData.Iv);
+            headers[EncryptionHeaders.KeyId] = encryptedData.KeyId;
+
+            headers = headers.Where(kvp => !string.IsNullOrEmpty(kvp.Value)).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
```

Comment:
I think you should just skip setting the `EncryptionHeaders.KeyId` header if `encryptedData.KeyId` is `null`. Doing this has the potential of changing other headers, which would be quite unexpected 😁

Reply:
Fix'd!